mesh-worker-service/src/main/java/io/functionmesh/compute/rest/api/SinksImpl.java

```diff
@@ -291,33 +309,121 @@ public SinkStatus getSinkStatus(
+                podList.getItems().forEach(pod -> {
+                    String name = pod.getMetadata() != null && pod.getMetadata().getName() != null ? pod.getMetadata().getName() : null;
+                    Integer shardId = name != null ? new Integer(name.substring(name.lastIndexOf("-"))) : null;
+                    if (shardId != null) {
+                        String address = KubernetesUtils.getServiceUrl(name, subdomain, nameSpaceName);
+                        channel[shardId] = ManagedChannelBuilder.forAddress(address, 9093)
+                                .usePlaintext()
+                                .build();
+                        stub[shardId] = InstanceControlGrpc.newFutureStub(channel[shardId]);
+
+                        SinkStatus.SinkInstanceStatus sinkInstanceStatus = sinkStatus.getInstances().get(shardId);
+                        if (sinkInstanceStatus != null) {
+                            SinkStatus.SinkInstanceStatus.SinkInstanceStatusData sinkInstanceStatusData = sinkInstanceStatus.getStatus();
+                            V1PodStatus podStatus = pod.getStatus();
+                            if(v1alpha1Sink.getSpec() != null && StringUtils.isNotEmpty(v1alpha1Sink.getSpec().getClusterName())) {
+                                sinkInstanceStatusData.setWorkerId(v1alpha1Sink.getSpec().getClusterName());
+                            }
+                            if(podStatus != null) {
+                                sinkInstanceStatusData.setRunning(podStatus.getPhase() != null && podStatus.getPhase().equals("Running"));
+                                if (podStatus.getContainerStatuses() != null && !podStatus.getContainerStatuses().isEmpty()) {
+                                    V1ContainerStatus containerStatus = podStatus.getContainerStatuses().get(0);
+                                    sinkInstanceStatusData.setNumRestarts(containerStatus.getRestartCount());
+                                }
+                            }
+                            // get status from grpc
+                            InstanceCommunication.FunctionStatus protoStatus = null;
+                            try {
+                                protoStatus = SinksUtil.getFunctionStatus(stub[shardId]).get(GRPC_TIMEOUT_SECS, TimeUnit.SECONDS);
+                            } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                                log.error("Get sink {}-{} status from grpc failed from namespace {}, error message: {}",
+                                        statefulSetName,
+                                        shardId,
+                                        nameSpaceName,
+                                        e.getMessage());
+                                sinkInstanceStatusData.setRunning(false);
+                                sinkInstanceStatusData.setError(e.getMessage());
+                            }
+                            if(protoStatus != null) {
```

Comment:
Looks this code block could be abstracted as a function which may be easier to test.